arch/xtensa/core/xtensa_intgen.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+import re
+import fileinput
+
+# Pass an Xtensa core-isa.h file on stdin or the command line, emits a
+# C file on output containing optimized interrupt dispatch routines.
+
```

Comment:
Suggest filing a user story to have someone with CMake chops make this part of the build.
I have no idea on how to do it either.